sale_order_retrieve_customer_after_delivery/__init__.py

```diff
@@ -0,0 +1,14 @@
+from . import models
+from odoo.tools import column_exists, create_column
+
+def pre_init_hook(env):
+   
+    if not column_exists(env.cr, "sale_order", "partner_shipping_2_id"):
+        create_column(env.cr, "sale_order", "partner_shipping_2_id", "character varying")
```

Comment:
This is wrong, `partner_shipping_2_id` should be an `integer` value, as a `Many2one` field:
```suggestion
        create_column(env.cr, "sale_order", "partner_shipping_2_id", "integer")
```